internal/stores/hostdb.go

```diff
@@ -181,3 +195,236 @@ func NewJSONHostDB(dir string) (*JSONHostDB, modules.ConsensusChangeID, error) {
 	}
 	return db, ccid, nil
 }
+
+// consensusInfoID defines the primary key of the entry in the consensusInfo
+// table.
+const consensusInfoID = 1
+
+type (
+	// SQLHostDB is a helper type for interacting with an SQL-based HostDB.
+	SQLHostDB struct {
+		staticDB *gorm.DB
+	}
+
+	// host defines a hostdb.Interaction as persisted in the DB.
+	host struct {
+		PublicKey     []byte         `gorm:"primaryKey"`
+		Announcements []announcement `gorm:"foreignKey:Host;references:PublicKey"`
+		Interactions  []interaction  `gorm:"foreignKey:Host;references:PublicKey"`
+	}
+
+	announcement struct {
+		ID          uint64 `gorm:"primaryKey"`
+		Host        []byte
+		BlockHeight uint64
+		BlockID     []byte
+		Timestamp   time.Time
+		NetAddress  string
+	}
+
+	// interaction defines a hostdb.Interaction as persisted in the DB.
+	interaction struct {
+		ID     uint64 `gorm:"primaryKey"`
+		Type   string
+		Result json.RawMessage
+
+		// Host and Timestamp form a composite index where Hosts has the
```

Comment:
I actually removed that again and just made it 2 separate indices. I think we might need some more research and insights before making fancy decisions like that. Because I think the hydrator can use a separate index on the foreign key but I'm not sure. about the combined index.

Reply:
`but I'm not sure about the combined index`

yeah so this was essentially my concern 👍

Reply:
`nit` remove comment